### ISSUE TITLE
Quality of life changes (Escape instant exit and back button)

### ISSUE THREAD

**Issue by MisterEpic** [VolmitSoftware/Adapt#193]
### Adding, Removing, or Changing

Changing

### Type of Modification

Quality of life improvements that will make the plugin smoother to use

### What are you trying to modify

I suggested some of these before, but I think the issue was closed.

1. Back button in GUI
Currently, clicking escape to go back closes and reopens the GUI briefly, which gets annoying fast. A button, such as a bed or other item which allows players to go back without the GUI quickly opening and closing would be great. My rationale behind this feature, is that once players become familiar with the plugin and skills, they likely will want to learn/unlearn skills quickly, and the delay between going back to each GUI will likely become quite frustrating.
![image](https://user-images.githubusercontent.com/111988642/190912785-be024dd2-3682-432e-ac65-d2889985dfe2.png)
![image](https://user-images.githubusercontent.com/111988642/190912791-c3cb2b76-cf8d-4100-a15b-af1c1aa005c5.png)

2. Escape key leaves entire GUI immediately
In conjuction with the first suggestion, making the escape key leave the entire GUI, even when in specific skill trees/abilities would also make use of the plugin more enjoyable. Players with higher ping will find this delay in leaving the GUI even more annoying. This delay may also prove problematic when in scenarios such as factions, where your base is being raided and you quickly need to change skills.
Players who also become very familiar with the skills/abilities, may also get frustrated when quickly switching/equipping skills, and being unable to leave the GUI immediately.

![thegif](https://user-images.githubusercontent.com/111988642/190912962-abfb1b42-efec-40dd-a5f9-c1ab3bfbf2eb.gif)

3. GUI closes and reopens when selecting a skill
Currently when selecting a skill (as demonstrated in the above gif), the GUI closes and reopens.
I won't go into detail on why this should be added, as the logic follows what i've already written in suggestion 1 and 2. If the GUI remained open without closing once upon selecting skills, I believe that would be a great addition.

4. Button to unequip/unlearn all skills
Once players enter the higher levels, keeping track of all the skill trees their power is invested in can get tough. I believe a button in the GUI like a barrier block, which when clicked unlearns all skills at once will greatly help this. Understandably, more abilities will be added in the future, and this will take up a GUI slot, so maybe a command could be used instead (e.g. /respec), or the GUI could be expanded into a double chest?
Simply put, I believe such a feature is important due to players potentially having multiple "builds". For example, players may choose to learn more hunter abilities/axe abilities when in combat, but prefer architect abilities/excavation abilities in base construction. Having an easy method of unlearning all abilities, instead of requiring players to manually go through and select each skill to unlearn, would be incredibly useful.

![image](https://user-images.githubusercontent.com/111988642/190912896-2a16e5d0-f91c-44d5-b7dc-444840b75941.png)

### Alternatives

No real alternative here, but maybe (since you don't like some of the GUI suggestions), make the suggested GUI changes a toggleable/configurable option?

**Comment by NextdoorPsycho** [comment]
I’m not doing this, I would have to close and reopen the GUI anyway if it’s slow that means your disk is slow Nobody else is having us issue

**Comment by TheekshanaA** [comment]
We have 2 nvme ssds running in raid 0. It is definitely not a slow disk.

**Comment by NextdoorPsycho** [comment]
You are the only one having this issue is it perhaps a plugin
incompatibility? try it on a server with just adapt and slowly add them
